.github/workflows/cd/core.yaml

```
@@ -0,0 +1,128 @@
+name: cd/core
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
+    paths:
+      - go.mod
+      - go.sum
+      - '**/*.go'
+
+jobs:
+  release-name:
+    runs-on: ubuntu-latest
+    outputs:
+      release-name: ${{ steps.release-name.outputs.release-name }}
+    steps:
+      - uses: actions/checkout@v3
+      - id: release-name
+        run: |
+          RELEASE_NAME=${{ startsWith(github.ref, 'refs/tags/') && github.ref_name || '$(git describe --tags)' }}
+          echo "::set-output name=release-name::${RELEASE_NAME#v}"
+
+  build:
+    runs-on: ubuntu-latest
+    needs: [release-name]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
```

```suggestion
          go-version: 1.18.5
```

Not necessary, but nice to update to the latest version while we are in the neighborhood

---

Is there a reason not to change to `1.19`?

---

This is pinning the minor version so it can freely upgrade the patch version. Right now, 1.8 and 1.8.5 are equivalent in this context but if/when 1.8.6 is release, using 1.8 will automatically update to 1.8.6 while 1.8.5 will not.

As for upgrading to 1.9, I haven't gone through the release in detail so not sure what changes, if any, are needed so it's out of scope of this PR.